docs/DEVELOPMENT.md

```diff
@@ -0,0 +1,402 @@
+- **gke** - This is the default profile for local development, which will be activated automatically with the `kubectl` context set to the default cluster of this GCP project.
+  - The corresponding kustomize YAML is in the `./components/<component_name>/kustomize/base` folder.
+- **gke-hpa** - This is the profile for building and deploying to the Prod environment, e.g. to a customer's Prod environment. Adds Horizontal Pod Autoscaler (HPA)
+  - The corresponding kustomize YAML is in the `./components/<component_name>/kustomize/hpa` folder.
+
+
+##  6. <a name='Debugging'></a>Debugging
+
+###  6.1. <a name='Debugginglocally'></a>Debugging locally
+
+####  6.1.1. <a name='Debuggingcomponentscommon'></a>Debugging `components/common`
+By default, VS Code will use the Python interpreter you've selected with the Python extensions (CMD + SHIFT + P -> __Select Interpreter__) so clicking __Debug__ and running without a configuration should work, so long as you have shifted the interpreter over and activated the "Common" VirtualEnv.
+
+Mileage will vary - you may need to create a "Debug Current File" Debug configuration in VS Code, particularly if you are in a multi-folder Workspace.
+
+####  6.1.2. <a name='Debuggingcomponentscomponent_name'></a>Debugging `components/<component_name>`
+This should also just work, so long as you have selected the right interpreter, are in the microservice folder, and have entered your VirtualEnv.
+
+Mileage will vary - you may need to create a "Debug Current File" Debug configuration in VS Code, particularly if you are in a multi-folder Workspace.
+
+###  6.2. <a name='DebuggingwithSkaffoldCloudCodeonGKEcluster'></a>5.2. Debugging with Skaffold + Cloud Code (on GKE cluster)
+> NOTE: You don't need a VirtualEnv for this option.
+
+First, install [Cloud Code](https://marketplace.visualstudio.com/items?itemName=GoogleCloudTools.cloudcode)
+
+* From Command Palette (CMD + SHIFT + P): __Cloud Code: Debug on Kubernetes__
+* Select the root `skaffold.yaml`
+* Run __All dependencies__ or the module that you want
+* Select a profile (dev for GKE)
+* Select context - the GKE cluster you're using
+* You will need to edit the `launch.json` file and add `/src` to the "sourceFileMap" field
+
+When you're done, make sure to fully disconnect the debugger, so it removes the running services.
+
+##  7. <a name='Testing'></a>Testing
+
+###  7.1. <a name='Unittesting'></a>Unit testing
+
+* Install Firebase CLI:
+  ```
+  curl -sL https://firebase.tools | bash
+  ```
+* Install Virtualenv and pip requirements
+  ```
+  # Start in the root folder
+  export BASE_DIR=$(pwd)
+
+  # Go to a specific microservice folder:
+  cd components/<component_name>
+  python3 -m venv .venv
+  source .venv/bin/activate
+  pip install -r requirements.txt
+  pip install -r requirements-test.txt
+
+  # If this component depends on the common folder:
+  pip install -r ../common/requirements.txt
```

Comment:
Requirements.txt for `common` should preferably be installed before the `microservice`

Reply:
Not in the local development. Each microservice builds image from the base_image, which is the common image. However, for local dev, you'll need to manually install those deps.